src/stratigraphy/main.py

```diff
@@ -513,28 +94,40 @@ def perform_matching(directory: Path, **params: dict) -> dict:
         mlflow.log_params(flatten(line_detection_params))
         mlflow.log_params(flatten(matching_params))
 
-    # instantiate all paths
-    input_directory = DATAPATH / "Benchmark"
-    ground_truth_path = input_directory / "ground_truth.json"
-    out_directory = input_directory / "evaluation"
-    predictions_path = input_directory / "extract" / "predictions.json"
     temp_directory = DATAPATH / "_temp"  # temporary directory to dump files for mlflow artifact logging
 
     # check if directories exist and create them when neccessary
     out_directory.mkdir(parents=True, exist_ok=True)
     temp_directory.mkdir(parents=True, exist_ok=True)
 
+    # if a file is specified instead of an input directory, copy the file to a temporary directory and work with that.
+    if input_directory.is_file():
+        if (temp_directory / "single_file").is_dir():
+            shutil.rmtree(temp_directory / "single_file")
+
+        Path.mkdir(temp_directory / "single_file")
+        shutil.copy(input_directory, temp_directory / "single_file")
+        input_directory = temp_directory / "single_file"
```

Comment:
I'm not a big fan of this temporary directory. Wouldn't it be possible to instead support passing a single file as well as a directory to the `perform_matching` and `evaluate_matching` methods.

Reply:
I thought of refactoring the code such that we loop over all files inside of the `start_pipeline` function. Then we can easily catch the case where we provide a file instead of a folder.

Also, the function calls would be nicer, as we specifically pass objects to the functions instead of paths. I believe that would make the code more robust, as we only had to check once if the files comply with what's expected. (which we're not doing at all at the moment).

What do you think about that suggestion?

Reply:
Sounds good, feel free to go ahead with that refactoring.